TYPO3-Solr/ext-solr
# [BUGFIX] Fix value resolution in SOLR_RELATION

# What this pr does

This changeset fixes issues with the resolution of values in SOLR_RELATION, fixed is the resolution of:
- foreignLabelFields (2nd level+)
- multiple values in m:n relations
- nested m:n relations
- nested multi values

Additionally the missing stdWrap functionality on values from mm relations is added.

Integration tests are added to ensure the right behaviour of SOLR_RELATION.

# How to test

Test scenarios are added in `RelationTest`

For manual testing:
-  Create a table A with an 1:m relation to table B
-  In table B create an n:m relation to table C
-  Configure indexing for table A with a multi value field that should contain all relations to table C
- Index the records
- Relations must be resolved correctly

Resolves: #3152
Resolves: #3408
